Tests/VaporTests/PasswordTests.swift

```diff
@@ -0,0 +1,111 @@
+import XCTVapor
+
+final class PasswordTests: XCTestCase {
+    func testSyncBCryptService() throws {
+        let test = Environment(name: "testing", arguments: ["vapor"])
+        let app = Application(test)
+        defer { app.shutdown() }
+        
+        let hash = try app.password.digest("vapor")
+        XCTAssertTrue(try BCryptDigest().verify("vapor", created: hash))
+        
+        let result = try app.password.verify("vapor", created: hash)
+        XCTAssertTrue(result)
+    }
+    
+    func testSyncPlaintextService() throws {
+        let test = Environment(name: "testing", arguments: ["vapor"])
+        let app = Application(test)
+        defer { app.shutdown() }
+        app.passwords.use(.plaintext)
+        
+        let hash = try app.password.digest("vapor")
+        XCTAssertEqual(hash, "vapor")
+        
+        let result = try app.password.verify("vapor", created: hash)
+        XCTAssertTrue(result)
+    }
+    
+    func testAsyncBCryptRequestPassword() throws {
+        let test = Environment(name: "testing", arguments: ["vapor"])
+        let app = Application(test)
+        defer { app.shutdown() }
+        
+        try assertAsyncRequestPasswordVerifies(.bcrypt, on: app)
+    }
+    
+    func testAsyncPlaintextRequestPassword() throws {
+        let test = Environment(name: "testing", arguments: ["vapor"])
+        let app = Application(test)
+        defer { app.shutdown() }
+        
+        try assertAsyncRequestPasswordVerifies(.plaintext, on: app)
+    }
+    
+    func testAsyncBCryptApplicationPassword() throws {
+        let test = Environment(name: "testing", arguments: ["vapor"])
+        let app = Application(test)
+        defer { app.shutdown() }
+        
+        try assertAsyncApplicationPasswordVerifies(.bcrypt, on: app)
+    }
+    
+    func testAsyncPlaintextApplicationPassword() throws {
+        let test = Environment(name: "testing", arguments: ["vapor"])
+        let app = Application(test)
+        defer { app.shutdown() }
+        
+        try assertAsyncApplicationPasswordVerifies(.plaintext, on: app)
+    }
+    
```

Comment:
Can you add one final test to ensure the test works 😆 Basically just to make sure the async password uses the requested provider
```suggestion
    
    func testAsyncUsesProvider() throws {
         let test = Environment(name: "testing", arguments: ["vapor"])
         let app = Application(test)
         defer { app.shutdown() }
         app.passwords.use(.plaintext)
         let hash = try app.password.async(on: app.threadPool, hopTo: app.eventLoopGroup.next()).digest("vapor").wait()
         XCTAssertEqual(hash, "vapor")
     }
     
```